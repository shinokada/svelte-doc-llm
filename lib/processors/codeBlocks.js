/**
 * Process code blocks - modify svelte example tags
 * @param {string} content - Markdown content
 * @returns {string} Processed content
 */
export function processCodeBlocks(content) {
  return content.replace(/^```svelte.*$/gm, '```svelte');
}

/**
 * Remove script section from markdown
 * @param {string} content - Markdown content
 * @returns {string} Processed content
 */
export function removeScriptSection(content) {
  // Remove script sections with optional attributes (lang, type, etc.)
  return content.replace(/<script[^>]*>[\s\S]*?<\/script>\s*/, '');
}
