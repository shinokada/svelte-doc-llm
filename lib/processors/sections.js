/**
 * Remove sections by title
 * @param {string} content - Markdown content
 * @param {Array<string>} ignoreList - List of section titles to remove
 * @returns {string} - Processed content
 */
export function removeSections(content, ignoreList) {
  if (!ignoreList.length) return content;
  
  let lines = content.split('\n');
  let processedLines = [];
  
  let insideIgnoredSection = false;
  let currentLevel = 0;
  
  for (let i = 0; i < lines.length; i++) {
    const line = lines[i];
    
    // Check if line is a heading
    const headingMatch = line.match(/^(#+)\s+(.+)$/);
    
    if (headingMatch) {
      const level = headingMatch[1].length;
      const title = headingMatch[2];
      
      // If we're ignoring a section and found a heading of same or higher level, stop ignoring
      if (insideIgnoredSection && level <= currentLevel) {
        insideIgnoredSection = false;
      }
      
      // Check if heading should be ignored
      if (ignoreList.includes(title) || ignoreList.includes(`${'#'.repeat(level)} ${title}`)) {
        insideIgnoredSection = true;
        currentLevel = level;
        continue;
      }
    }
    
    if (!insideIgnoredSection) {
      processedLines.push(line);
    }
  }
  
  return processedLines.join('\n');
}