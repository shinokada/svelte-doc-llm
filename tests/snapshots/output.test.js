import { describe, it, expect } from 'vitest';
import { processCodeBlocks } from '../../lib/processors/codeBlocks.js';
import { removeSections } from '../../lib/processors/sections.js';

describe('output snapshots', () => {
  it('should match processed code block snapshot', () => {
    const input = `
\`\`\`svelte example hideScript
<script>
  let count = 0;
</script>

<button on:click={() => count++}>
  Count: {count}
</button>
\`\`\`
`;
    expect(processCodeBlocks(input)).toMatchSnapshot();
  });

  it('should match section removal snapshot', () => {
    const input = `
# Main Title

## Introduction
Some intro text

## Installation
npm install package

## Usage
How to use

## API
API documentation
`;
    const result = removeSections(input, ['Installation']);
    expect(result).toMatchSnapshot();
  });
});