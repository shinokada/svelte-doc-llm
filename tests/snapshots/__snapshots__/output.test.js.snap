// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`output snapshots > should match processed code block snapshot 1`] = `
"
\`\`\`svelte
<script>
  let count = 0;
</script>

<button on:click={() => count++}>
  Count: {count}
</button>
\`\`\`
"
`;

exports[`output snapshots > should match section removal snapshot 1`] = `
"
# Main Title

## Introduction
Some intro text

## Usage
How to use

## API
API documentation
"
`;
